sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -716,4 +717,118 @@ public HttpPipeline getHttpPipeline() {
+    }
+
+    <T> Mono<DigitalTwinsResponse<T>> getComponentWithResponse(String digitalTwinId, String componentPath, Class<T> clazz, Context context) {
+        return protocolLayer.getDigitalTwins().getComponentWithResponseAsync(digitalTwinId, componentPath, context)
+            .flatMap(response -> {
+                T genericResponse = mapper.convertValue(response.getValue(), clazz);
+                DigitalTwinsResponseHeaders twinHeaders = mapper.convertValue(response.getDeserializedHeaders(), DigitalTwinsResponseHeaders.class);
+                return Mono.just(new DigitalTwinsResponse<T>(response.getRequest(), response.getStatusCode(), response.getHeaders(), genericResponse, twinHeaders));
+            });
+    }
+
+    /**
+     * Patch a component on a digital twin.
+     * @param digitalTwinId The Id of the digital twin that has the component to patch.
+     * @param componentPath The path of the component on the digital twin.
+     * @param componentUpdateOperations The application json patch to apply to the component. See {@link com.azure.digitaltwins.core.util.UpdateOperationUtility} for building
+     *                                  this argument.
+     * @return An empty Mono.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> updateComponent(String digitalTwinId, String componentPath, List<Object> componentUpdateOperations) {
+        return updateComponentWithResponse(digitalTwinId, componentPath, componentUpdateOperations, new UpdateComponentRequestOptions())
+            .flatMap(response -> Mono.justOrEmpty(response.getValue()));
+    }
+
+    /**
+     * Patch a component on a digital twin.
+     * @param digitalTwinId The Id of the digital twin that has the component to patch.
+     * @param componentPath The path of the component on the digital twin.
+     * @param componentUpdateOperations The application json patch to apply to the component. See {@link com.azure.digitaltwins.core.util.UpdateOperationUtility} for building
+     *                                  this argument.
+     * @param requestOptions The optional parameters for this request.
+     * @return A {@link DigitalTwinsResponse} containing an empty Mono.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DigitalTwinsResponse<Void>> updateComponentWithResponse(String digitalTwinId, String componentPath, List<Object> componentUpdateOperations, UpdateComponentRequestOptions requestOptions) {
+        return withContext(context -> updateComponentWithResponse(digitalTwinId, componentPath, componentUpdateOperations, requestOptions, context));
+    }
+
+    Mono<DigitalTwinsResponse<Void>> updateComponentWithResponse(String digitalTwinId, String componentPath, List<Object> componentUpdateOperations, UpdateComponentRequestOptions requestOptions, Context context) {
+        String ifMatch = requestOptions != null ? requestOptions.getIfMatch() : null;
```

Comment:
Good point, I believe I've missed this check in the update relationship API, would you mind adding this there as well?

Reply:
Sure thing

Reply:
I'll do the same for the deleteRelationship API as well, since it is missing there, too